sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts

```diff
@@ -134,37 +144,168 @@ export class IER12ProcessingService {
+        institutionProgramCode: educationProgram.institutionProgramCode,
+        programCompletionYears: educationProgram.completionYears,
+        studyStartDate: new Date(offering.studyStartDate),
+        studyEndDate: new Date(offering.studyEndDate),
+        tuitionFees: offering.actualTuitionCosts,
+        booksAndSuppliesCost: offering.programRelatedCosts,
+        mandatoryFees: offering.mandatoryFees,
+        exceptionExpenses: offering.exceptionalExpenses,
+        totalFundedWeeks: pendingAssessment.assessmentData.weeks,
+        applicationSubmittedDate: application.submittedDate,
+        programYear: applicationProgramYear.programYear,
+        applicationStatus: application.applicationStatus,
+        applicationStatusDate: application.applicationStatusUpdatedOn,
+        assessmentAwards,
+        hasProvincialDefaultRestriction,
+        hasProvincialOveraward: studentOverawardsBalance
+          ? studentOverawardsBalance[FullTimeAwardTypes.BCSL] > 0
+          : false,
+        hasFederalOveraward: studentOverawardsBalance
+          ? studentOverawardsBalance[FullTimeAwardTypes.CSLF] > 0
+          : false,
+        hasRestriction,
+        assessmentTriggerType: pendingAssessment.triggerType,
+        scholasticStandingChangeType: scholasticStanding?.changeType,
+        assessmentDate: pendingAssessment.assessmentDate,
+        hasPartner,
+        parentalAssets:
+          pendingAssessment.workflowData.calculatedData.parentalAssets,
+        coeStatus: disbursement.coeStatus,
+        disbursementScheduleStatus: disbursement.disbursementScheduleStatus,
+        earliestDateOfDisbursement: new Date(disbursement.disbursementDate),
+        dateOfDisbursement: disbursementReceipt?.disburseDate
+          ? new Date(disbursementReceipt.disburseDate)
+          : null,
+        disbursementCancelDate:
+          disbursement.disbursementScheduleStatus ===
+          DisbursementScheduleStatus.Cancelled
+            ? disbursement.updatedAt
+            : null,
+        disbursementAwards: this.getDisbursementAwards(
+          disbursement.disbursementValues,
+        ),
+      };
+      ier12Records.push(ier12Record);
+    }
+    return ier12Records;
+  }
+
+  /**
+   * Get all disbursement award details for IER.
+   * @param disbursementValues disbursement values.
+   * @returns award details.
+   */
+  private getDisbursementAwards(
+    disbursementValues: DisbursementValue[],
+  ): IERAward[] {
+    return disbursementValues.map<IERAward>((disbursementValue) => ({
+      valueType: disbursementValue.valueType,
+      valueCode: disbursementValue.valueCode,
+      valueAmount: disbursementValue.valueAmount,
+    }));
+  }
+
+  /**
+   * Get all disbursement award details for IER of all disbursements that belong
+   * to a given assessment.
+   * @param disbursementSchedules disbursement schedules.
+   * @returns assessment award details.
+   */
+  private getAssessmentAwards(
```

Comment:
Is the idea of this method to concat all the awards from the disbursements?
If yes I believe that it is equivalent to below:
```ts
const awards = disbursementSchedules.flatMap(
  (disbursementSchedule) => disbursementSchedule.disbursementValues,
);
```